src/language-js/comments/handle-comments.js

```diff
@@ -887,6 +888,44 @@ function handleSwitchDefaultCaseComments({
   return true;
 }
 
+/**
+ * Handle `Comment2` and `Comment4`.
+ *
+ *   type Foo = (
+ *     | "thing1" // Comment1
+ *     | "thing2" // Comment2
+ *   )[];
+ *
+ *   type Foo = (
+ *     | "thing1" // Comment3
+ *     | "thing2" // Comment4
+ *   ) & Bar;
+ *
+ * @param {CommentContext} context
+ * @returns {boolean}
+ */
+function handleLastUnionElementInExpression({
+  comment,
+  precedingNode,
+  enclosingNode,
+  followingNode,
+}) {
+  if (
+    precedingNode &&
+    (precedingNode.type === "TSUnionType" ||
+      precedingNode.type === "UnionTypeAnnotation") &&
```

Comment:
I don't think tests are using flow parsers. Please add more parsers in the tests.

Reply:
Thank you, added [9cd628f](https://github.com/prettier/prettier/pull/15409/commits/9cd628f95046a0bc544783331d5313272abc8195)